core/src/main/scala/org/broadinstitute/workbench/hamm/db/WorkflowCostDAO.scala

```diff
@@ -0,0 +1,100 @@
+package org.broadinstitute.workbench.hamm
+package db
+
+import java.time.Instant
+import cats.effect.Async
+import cats.implicits._
+import doobie._
+import doobie.free.connection
+import doobie.implicits._
+import org.broadinstitute.workbench.hamm.db.WorkflowCostDAO._
+
+class WorkflowCostDAO[F[_]: Async](transactor: Transactor[F]) {
+  val createTable: F[Unit] = {
+    val connIO = for {
+      createTable <- createSql.run
+      createIndex <- if(createTable == 0) createIndex.run else connection.raiseError(new Exception(s"creating table $workflowTableName failed with $createTable"))
+      _ <- if(createIndex == 0) ().pure[ConnectionIO] else connection.raiseError(new Exception(s"creating index labelsIndex failed with $createIndex"))
+    } yield ()
+
+    connIO.transact(transactor)
+  }
+
+  def insert(workflowddb: WorkflowDB): F[Int] = insertWorkflowSql(workflowddb).run.transact[F](transactor)
+
+  def getWorkflowDB(workflowId: WorkflowId): F[WorkflowDB] = getWorkflowDBSql(workflowId).unique.transact[F](transactor)
+
+  def getWorkflowCostWithLabel(label: Label): F[Double] = getWorkflowCostSqlWithLabel(label).unique.transact[F](transactor)
+}
+
+object WorkflowCostDAO {
+  def apply[F[_]: Async](transactor: Transactor[F]): WorkflowCostDAO[F] = new WorkflowCostDAO(transactor)
+
+  val createSql: Update0 =
+    (fr"CREATE TABLE IF NOT EXISTS" ++ workflowTableName ++
+      fr"(id SERIAL NOT NULL," ++
+      workflowIdFragment ++ fr"UUID NOT NULL," ++
+      parentWorkflowIdFragment ++ fr"UUID," ++
+      rootWorkflowIdFragment ++ fr"UUID," ++
+      isSubWorkflowFragment ++ fr"BOOLEAN NOT NULL," ++
+      startTimeFragment ++ fr"TIMESTAMPTZ NOT NULL," ++
+      endTimeFragment ++ fr"TIMESTAMPTZ NOT NULL," ++
+      labelsFragment ++ fr"JSONB," ++
+      costFragment ++ fr"FLOAT8 NOT NULL" ++
+      fr")").update
```

Comment:
I think we discussed having workflow collections as a separate column (it's a label, but it's special because that's what we're using to authorize)